### PR TITLE
mobile: removing main interface

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by alyssawilk** [comment]
cc @fredyw this is the PR you were hoping for last time =P

File: mobile/library/common/jni/jni_impl.cc [hunk 1]
```diff
@@ -962,8 +962,8 @@ Java_io_envoyproxy_envoymobile_engine_JniLibrary_registerKeyValueStore(JNIEnv* e
   api->remove = jvm_kv_store_remove;
   api->context = retained_context;
 
-  envoy_status_t result = register_platform_api(env->GetStringUTFChars(name, nullptr), api);
-  return result;
+  Envoy::Api::External::registerApi(env->GetStringUTFChars(name, nullptr), api);
```
Comment (fredyw) [inline mobile/library/common/jni/jni_impl.cc:965]: I might have missed updating this code to use `JniHelper`. Can we use `JniHelper::getStringUtfChars`? https://github.com/envoyproxy/envoy/blob/226d37da07b1f015827b44f1dc093ec81d71b063/mobile/library/common/jni/jni_helper.cc#L65

File: mobile/library/common/jni/jni_impl.cc [hunk 2]
```diff
@@ -997,8 +997,8 @@ Java_io_envoyproxy_envoymobile_engine_JniLibrary_registerFilterFactory(JNIEnv* e
   api->static_context = retained_context;
   api->instance_context = nullptr;
 
-  envoy_status_t result = register_platform_api(env->GetStringUTFChars(filter_name, nullptr), api);
-  return result;
+  Envoy::Api::External::registerApi(env->GetStringUTFChars(filter_name, nullptr), api);
```
Comment (fredyw) [inline mobile/library/common/jni/jni_impl.cc:1000]: Use `JniHelper::getStringUtfChars`

File: mobile/library/common/jni/jni_impl.cc [hunk 3]
```diff
@@ -1119,9 +1119,9 @@ Java_io_envoyproxy_envoymobile_engine_JniLibrary_registerStringAccessor(JNIEnv*
   string_accessor->get_string = jvm_get_string;
   string_accessor->context = retained_context;
 
-  envoy_status_t result =
-      register_platform_api(env->GetStringUTFChars(accessor_name, nullptr), string_accessor);
-  return result;
+  Envoy::Api::External::registerApi(env->GetStringUTFChars(accessor_name, nullptr),
```
Comment (fredyw) [inline mobile/library/common/jni/jni_impl.cc:1122]: Use `JniHelper::getStringUtfChars`

**Review by fredyw — APPROVED** [review]
Nice!

**Review by abeyad — APPROVED** [review]

**Review by abeyad — APPROVED** [review]

**Pull request merged**